--- qt5-build.eclass.orig	2022-04-20 21:42:30.967568052 -0400
+++ qt5-build.eclass	2022-04-20 21:38:52.497584358 -0400
@@ -134,7 +134,7 @@
 		SLOT=5/$(ver_cut 1-2) ;;
 esac
 
-IUSE="debug test"
+IUSE="cpu_flags_x86_sse2 debug test"
 
 if [[ ${QT5_BUILD_TYPE} == release ]]; then
 	RESTRICT="test" # bug 457182
@@ -550,6 +550,9 @@
 		$(is-flagq -mno-dsp   && echo -no-mips_dsp)
 		$(is-flagq -mno-dspr2 && echo -no-mips_dspr2)
 
+		# Check for availability of SSE2 and switch off if necessary
+		$(use cpu_flags_x86_sse2 || echo -no-sse2)
+
 		# use pkg-config to detect include and library paths
 		-pkg-config
 
