--- qt5-build.eclass.orig	2020-08-15 16:29:52.260751973 +0200
+++ qt5-build.eclass	2020-08-15 16:29:52.259751818 +0200
@@ -106,7 +106,7 @@
 # Build directory for out-of-source builds.
 : ${QT5_BUILD_DIR:=${S}_build}
 
-IUSE="debug test"
+IUSE="cpu_flags_x86_sse2 debug test"
 
 [[ ${QT5_BUILD_TYPE} == release ]] && RESTRICT+=" test" # bug 457182
 
@@ -550,6 +550,9 @@
 		$(is-flagq -mno-dsp   && echo -no-mips_dsp)
 		$(is-flagq -mno-dspr2 && echo -no-mips_dspr2)
 
+		# Check for availability of SSE2 and switch off if necessary
+		$(use cpu_flags_x86_sse2 || echo -no-sse2)
+		
 		# use pkg-config to detect include and library paths
 		-pkg-config
 
